setup/setup.js

```diff
@@ -1,93 +1,182 @@
+                                    }
+                                }, (err, response) => {
+
+                                    if (err) {
+                                        callback('Could not create new PubNub after-publish Event Handler. Please contact support@pubnub.com.');
+                                    }
+
+                                    status('Creating new after-publish Event Handler...');
+
+                                    $.get('functions/get-kv-state.js', (code) => {
+
+                                        api.request('post', ['api', 'v1', 'blocks', 'key', key.id, 'event_handler'], {
+                                            data: {
+                                                key_id: key.id,
+                                                block_id: block.id,
+                                                type: 'js',
+                                                event: 'js-after-publish',
+                                                channels: 'global',
+                                                name: 'get-kv-state',
+                                                code: code,
+                                                output: 'output-0.5823105682419438'
+                                            }
+                                        }, (err, response) => {
+
+                                            if (err) {
+                                                callback('Could not create new PubNub after-publish Event Handler. Please contact support@pubnub.com.');
+                                            }
+
+                                            status('Starting Pubnub Function...');
+
+                                            api.request('post', ['api', 'v1', 'blocks', 'key', key.id, 'block', block.id, 'start'], {
+                                                data: {
+                                                    block_id: block.id,
+                                                    key_id: key.id,
+                                                    action: 'start'
+                                                }
+                                            }, (err, response) => {
+
+                                                if (err) {
+                                                    callback('Could not start PubNub Function. Please contact support@pubnub.com.');
+                                                }
+
+                                                status('Done!');
+
+                                                callback(null, { pub: key.publish_key, sub: key.subscribe_key });
+                                            });
+                                        });
+                                    }, 'text');
+                                });
+                            }, 'text');
+                        });
+                    });
```

Comment:
async/series can help with the nested callbacks:
https://caolan.github.io/async/docs.html#series

Reply:
I was intending to update to node 8 and using native async/await like I did [here](https://github.com/pubnub/pubnub-open-growth/blob/master/deploy) but didn't feel like figuring out how to plug it into your UI so just continued your pattern

Reply:
o(╥﹏╥)o

Reply:
I'm so sorry, I know it's all my fault for hacking together too fast

Reply:
Don't be, it works as intended with minimal effort for now. After MVP I'll take a few days and write a nice clean fully featured version.

Reply:
I'm just being overdramatic for fun :)

http://icompile.eladkarako.com/wp-content/uploads/2016/01/icompile.eladkarako.com_callback_hell.gif